application/controllers/rest/V1/QuestionGroupController.php

```diff
@@ -0,0 +1,426 @@
+     *              }
+     *          )
+     *     ),
+     *     @OA\Response(
+     *          response="400",
+     *          description="Bad request"
+     *     ),
+     *     @OA\Response(
+     *          response="401",
+     *          description="Unauthorized"
+     *     ),
+     *      @OA\Response(
+     *          response="404",
+     *          description="Not found"
+     *     )
+     * )
+     *
+     * @param string $id Question Group Id
+     * @return void
+     */
+    public function actionIndexPut($id)
+    {
+        $request = Yii::app()->request;
+        $data    = $request->getRestParams();
+
+        $requestData = [
+            'sessionKey' => $this->getAuthToken(),
+            'groupID' => $id,
+            'groupData' => array(
+                'group_name' => isset($data['group_name']) ? $data['group_name'] : '',
+                'description' => isset($data['description']) ? $data['description'] : '',
+                'language' => isset($data['language']) ? $data['language'] : '',
+                'questiongroupl10ns' => isset($data['questiongroupl10ns']) ? $data['questiongroupl10ns'] : '',
+                'group_order' => isset($data['group_order']) ? $data['group_order'] : '',
+                'randomization_group' => isset($data['randomization_group']) ? $data['randomization_group'] : '',
+                'grelevance' => isset($data['grelevance']) ? $data['grelevance'] : ''
+            )
+        ];
+        $commandResponse = (new QuestionGroupPropertiesSet())
+            ->run(new Request($requestData));
+
+        $this->renderCommandResponse($commandResponse);
+    }
+
+    /**
+     * Delete question group by id.
+     *
+     * @OA\Delete(
+     *      path="/rest/v1/questionGroup/{id}",
+     *      security={{"bearerAuth":{}}},
+     *      summary="Delete question group by id",
+     *      description="Delete question group by id",
+     *      tags={"Question Group"},
+     *      @OA\Parameter(
+     *          description="Question group id",
+     *          in="path",
+     *          name="id",
+     *          required=true,
+     *          @OA\Schema(type="string"),
+     *      ),
+     *      @OA\Response(
+     *          response="200",
+     *          description="Success"
+     *      ),
+     *      @OA\Response(
+     *          response="400",
+     *          description="Bad request"
+     *      ),
+     *      @OA\Response(
+     *          response="401",
+     *          description="Unauthorized"
+     *      ),
+     *      @OA\Response(
+     *          response="404",
+     *          description="Not found"
+     *     )
+     * )
+     *
+     * @param string $id Question Group Id
+     * @return void
+     */
+    public function actionIndexDelete($id)
+    {
+        $requestData = [
+            'sessionKey' => $this->getAuthToken(),
+            'groupID' => $id
+        ];
+        $commandResponse = (new QuestionGroupDelete())
```

Comment:
@fosteruk Do we need to hardcode the command objects? Would reflection or similar be an alternative?